segmentation.py

```diff
@@ -38,11 +38,15 @@ def populate_component_occupancy(schematic: PangenomeSchematic):
 
 
 def populate_component_matrix(paths: List[Path], schematic: PangenomeSchematic):
-    for component in schematic.components:
-        # paths paths are in the same order as schematic.path_names
-        for i, path in enumerate(paths):
-            relevant = [bin for bin in path.bins if
-                        component.first_bin <= bin.bin_id <= component.last_bin]  # very costly loop
+    # the loops are 1) paths, and then 2) schematic.components
+    # paths are in the same order as schematic.path_names
+    for i, path in enumerate(paths):
+        sortedBins = SortedDict((bin.bin_id, bin) for bin in path.bins)
```

Comment:
Nitpick: please use snake_case like in the rest of the codebase

Reply:
got it, will do. It's a habit from the Java world